### ISSUE TITLE
Set up text editor

### ISSUE THREAD

**Issue by ijmccallum** [WikiLogic/Server#212]
Had a look through some options, this one looks pretty good: https://github.com/Alex-D/Trumbowyg

**Comment by ijmccallum** [comment]
In fact - I should put together a requirements table and run through more editors to find the best suited, more work but probably worth it!
Details to go into this page: https://github.com/WikiLogic/WikiLogic/wiki/Editor

**Comment by DouglasUT2018** [comment]
Just to check - Sunday at 10am for meeting? Was going to text you now 
but its late for you and on a school night!

The editor can be extremely simple plain text. Cant think of any 
punctuation or styles (bold, italics, etc) they would need. Might be 
worth exploring the rest of the UI and getting back to this.

**Comment by ijmccallum** [comment]
added some comments! http://www.wikilogicfoundation.org/wiki/index.php?title=UI#Critique_of_current_layout
Got to specify which time zone you're meaning! 10amUS / 3pmUK - that would work!
Yeah - no bold / italic / etc. I've added an editor plugin as it would allow us to skip writing any code to manage text input directly. My plan is to remove the normal buttons (already done!) and add our own, eg the fallacy tagger or linking dictionary terms. If the tagger is only applying to part of a statement we need a way to manage which part. The editor I've got turns the text of a claim into a much more easily handled data structure, as opposed to reading it off the DOM (which is a nightmare). Also, the editor will let us turn editing on/off without having to play around with hacking html input's (which is what i was doing before).
Massive plus to the editor I've got - it allows us to extend to different languages :D I'm not looking at that in detail yet - just a nod to the future, might be able to hook into the way it does that and use it to power the language used throughout the rest of the UI. We will have to build our own eventually, but that'll be a bit down the road.

Date: Wed, 22 Jun 2016 00:00:57 -0700
